banyand/metadata/schema/etcd.go

```diff
@@ -402,15 +534,49 @@ func (e *etcdSchemaRegistry) get(ctx context.Context, key string, message proto.
 	return nil
 }
 
-func (e *etcdSchemaRegistry) update(ctx context.Context, group *commonv1.Group, key string, message proto.Message) error {
-	val, err := proto.Marshal(message)
+func (e *etcdSchemaRegistry) update(ctx context.Context, group *commonv1.Group, res *resource, notification bool) error {
+	val, err := proto.Marshal(res.Message)
 	if err != nil {
 		return err
 	}
-	_, err = e.kv.Put(ctx, key, string(val))
+	_, err = e.kv.Put(ctx, res.Key(), string(val))
 	if err != nil {
 		return err
 	}
+
+	if notification {
+		switch res.typ {
+		case ResourceStream:
+			e.onEntityUpdate(commonv1.Catalog_CATALOG_STREAM, res.GetMetadata())
+		case ResourceMeasure:
+			e.onEntityUpdate(commonv1.Catalog_CATALOG_MEASURE, res.GetMetadata())
+		case ResourceIndexRuleBinding:
+			rb := res.Message.(*databasev1.IndexRuleBinding)
+			e.onEntityUpdate(rb.GetSubject().GetCatalog(), &commonv1.Metadata{
+				Name:  rb.GetSubject().GetName(),
+				Group: group.GetName(),
+			})
+		case ResourceIndexRule:
+			for _, ownerRef := range res.GetMetadata().GetOwnerReferences() {
+				rb, innerErr := e.GetIndexRuleBinding(ctx, &commonv1.Metadata{
+					Name:  ownerRef.GetName(),
+					Group: group.GetName(),
+				})
+
+				if innerErr != nil {
+					// TODO: possible inconsistent state?
```

Comment:
The handler should be idempotent. The notifier has to retry the handler till it returns no error.